### PR TITLE
Beginning of 2G quantum repeater in QUISP

### PR DESCRIPTION
**Pull request by PorametPat**
- Add markdown for 2G repeater protocol
- Add figures for 2G repeater protocol markdown

### CODE REVIEW

**Comment by zigen** [comment]
@PorametPat could you rebase or merge the latest master branch? there's a lot of annoying diffs so it's hard to review...

**Review by PorametPat — COMMENTED** [review]
Added code and file should be fine now

**Review by PorametPat — COMMENTED** [review]
Conflict solved

File: .gitignore [hunk 1]
```diff
@@ -207,17 +207,7 @@ $RECYCLE.BIN/
 *.out
 *.app
 
-
-### https://raw.github.com/github/gitignore/218a941be92679ce67d0484547e3e142b2f5f6f0/Global/VisualStudioCode.gitignore
```
Comment (zigen) [inline .gitignore:211]: We want to share these files with the project members. Could you add it?

File: .gitignore [hunk 2]
```diff
@@ -207,17 +207,7 @@ $RECYCLE.BIN/
...
-!.vscode/launch.json
-!.vscode/extensions.json
-*.code-workspace
-
-# Local History for Visual Studio Code
-.history/
-
+inet/
+quisp/.vscode/
+quisp/modules/.vs/
```
Comment (zigen) [inline .gitignore:212]: what is the `.vs` directory? do we need to exclude from git management?

File: quisp/modules/QNIC/StationaryQubit/logicalStationaryQubit.h [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+                // Logical information
+                std::vector<StationaryQubit*> encoded_qubits_list;
+                logicalStationaryQubit *logical_entangled_partner = nullptr; 
+
+                VectorXcd ideal_logical_Bell_pair;
+                MatrixXcd ideal_logical_density_matrix;
+            
+            protected: 
+                virtual void initialize();
```
Comment (zigen) [inline quisp/modules/QNIC/StationaryQubit/logicalStationaryQubit.h:68]: if there's no class that inherits the `LogicalStationaryQubit` class, you don't need to add `virtual`. remove it.

File: quisp/modules/QNIC/StationaryQubit/logicalStationaryQubit.h [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            MatrixXcd state_in_density_matrix;
+            VectorXcd state_in_ket;
+        } logical_quantum_state;
+/*
+        typedef struct _measurement_outcome{
+            char basis;
+            bool outcome_is_plus;
+        }measurement_outcome;
+*/
+        typedef struct _logical_measurement_operator{
```
Comment (zigen) [inline quisp/modules/QNIC/StationaryQubit/logicalStationaryQubit.h:29]: you don't need to use `typedef ~` style.
it's ok like this
```cpp
struct LogicalMeasurementOperator {
  MatrixXcd plus;            
  MatrixXcd minus;            
  VectorXcd plus_ket;            
  VectorXcd minus_ket;            
  char basis;
} ;
```
https://godbolt.org/z/aberedT71

File: .gitignore (hunk 2 above)
Comment (zigen) [reply .gitignore:214]: hmm, I've never seen this. we don't need it. could you remove it?

File: .gitignore (hunk 1 above)
Comment (zigen) [reply .gitignore:211]: I wanted to say, we need this and the following lines you deleted.

File: .gitignore (hunk 2 above)
Comment (PorametPat) [reply .gitignore:214]: yes

**Comment by zigen** [comment]
ah the `.vs` directory comes from VisualStudio. @PorametPat you're using VisualStudio?

File: .gitignore (hunk 1 above)
Comment (zigen) [reply .gitignore:211]: no, just it has deleted.
https://github.com/sfc-aqua/quisp/pull/182/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947

**Comment by zigen** [comment]
I recommend adding [gitignore boilerplate](https://github.com/github/gitignore/blob/master/VisualStudio.gitignore) to the `.gitignore` for your development environment

**Comment by PorametPat** [comment]
Yes, I just switch to use Visual Studio instead of VS code to use `clang-format`.

File: quisp/modules/QNIC/StationaryQubit/LogicalBellSteane713.txt [hunk 5]
```diff
@@ -0,0 +1,16384 @@
+8.838834764831846047e-02 0.000000000000000000e+00
```
Comment (zigen) [inline quisp/modules/QNIC/StationaryQubit/LogicalBellSteane713.txt:1]: could you remove this file?

**Review by zigen — CHANGES_REQUESTED** [review]
could you remove `.vs` dir from git?
you can use this `$ git rm --cached -r quisp/.vs`

File: quisp/modules/QNIC/StationaryQubit/LogicalBellSteane713.txt (hunk 5 above)
Comment (PorametPat) [reply quisp/modules/QNIC/StationaryQubit/LogicalBellSteane713.txt:1]: That file contains vector of logical qubit state that needed when we do a measurement for state-tomography. So we cannot remove that.

Comment (zigen) [reply quisp/modules/QNIC/StationaryQubit/LogicalBellSteane713.txt:1]: ah here. I got it. 
https://github.com/sfc-aqua/quisp/pull/182/files#diff-268b7108d5342006616abc1242bfea24abcb899da935b1002706159301ceb7b2R255

**Comment by zigen** [comment]
Yeah, it's hard to merge it. Why don't you divide this into small parts? I think we can merge your markdown files at least.

**Comment by PorametPat** [comment]
Okay

**Comment by zigen** [comment]
let's close it. we already had a lot of comments so I don't want to have more discussion on this PR.

**Pull request closed without merging**